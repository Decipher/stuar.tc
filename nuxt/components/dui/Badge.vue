<template>
  <div :class="classes"><slot /></div>
</template>

<script>
export default {
  props: {
    size: {
      type: String,
      default: undefined,
    },
    type: {
      type: String,
      default: undefined,
    },
  },
  computed: {
    classes: ({ size, type }) =>
      ['badge', type && `badge-${type}`, size && `badge-${size}`].filter(
        (o) => o
      ),
  },
}
</script>
