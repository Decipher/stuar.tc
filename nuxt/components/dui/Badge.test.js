import { mount } from '@vue/test-utils'
import Badge from './Badge.vue'

describe('Badge', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Badge)
    expect(wrapper.vm).toBeTruthy()

    // Expect HTML to match snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })
})
