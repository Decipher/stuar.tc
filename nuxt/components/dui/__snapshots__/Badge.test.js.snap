// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge is a Vue instance 1`] = `"<div class=\\"badge\\"></div>"`;
